Improved: OFBiz audit/investigation (OFBIZ-12391)

PierreSmits opened the pull request:
modified:
ModelEntity.java - adding fields for auditing/investigating record creation and modification by default
to entities when OFBiz is initialised:
 * createdDate
 * createdByUserLogin
 * lastModifiedDate
 * lastModifiedByUserLogin

PierreSmits committed:
Improved: OFBiz audit/investigation (OFBIZ-12391)

modified:
ModelEntity.java - adding fields for auditing/investigating record creation and modification by default
to entities when OFBiz is initialised:
 * createdDate
 * createdByUserLogin
 * lastModifiedDate
 * lastModifiedByUserLogin

mbrohl commented:
AFAIK this can all be configured in the entity model as I mentioned in another PR some days ago.
Why is this change necessary?

PierreSmits commented:
Thank you for your feedback, Michael,

I am confident that you mentioned something somewhere about the aspect that something could be done somewhere else in OFBiz, but without actual reference to where this happened it hard to check whether it is pertaining to changes under this PR and/or related to the ticket associated. It could even be imaginary (not saying that it is).

So please provide the reference (preferably as a link), and if those comment there/then do not explicitly relate to the changes under this PR please also provide reasoning why it should.

It will surely help other contributors to form an opinion on whether having this in the code base, or not.

PierreSmits committed:
improved: Trustworthy OFBiz (OFBIZ-12391)

modified: removed fields from entity definitions for accounting component brought in from ModelEntity.java

PierreSmits committed:
improved: Trustworthy OFBiz - content entities (OFBIZ-12391)

modified: removed fields from entity definitions for content component brought in from ModelEntity.java

PierreSmits committed:
improved: Trustworthy OFBiz - humanres entities (OFBIZ-12391)

modified: removed fields from entity definitions for humanres component brought in from ModelEntity.java

PierreSmits committed:
improved: Trustworthy OFBiz - marketing entities (OFBIZ-12391)

modified: removed fields from entity definitions for marketing component brought in from ModelEntity.java

PierreSmits committed:
improved: Trustworthy OFBiz - order entities (OFBIZ-12391)

modified: removed fields from entity definitions for order component brought in from ModelEntity.java

PierreSmits committed:
improved: Trustworthy OFBiz - party entities (OFBIZ-12391)

modified: removed fields from entity definitions for party component brought in from ModelEntity.java

PierreSmits committed:
improved: Trustworthy OFBiz - product entities (OFBIZ-12391)

modified: removed fields from entity definitions for product component brought in from ModelEntity.java

PierreSmits committed:
improved: Trustworthy OFBiz - shipment entities (OFBIZ-12391)

modified: removed fields from entity definitions for facility component brought in from ModelEntity.java

PierreSmits committed:
improved: Trustworthy OFBiz - workeffort entities (OFBIZ-12391)

modified: removed fields from entity definitions for workeffort component brought in from ModelEntity.java

PierreSmits committed:
improved: Trustworthy OFBiz - service entities (OFBIZ-12391)

modified: removed fields from entity definitions for service component brought in from ModelEntity.java

PierreSmits committed:
improved: Trustworthy OFBiz - opportunity (OFBIZ-12391)

modified:
createSalesForecast - adjusted function to change setting entity fields re creation and modification
updateSalesForecast - adjusted function to change setting entity fields re modification
saveSalesForecastHistory - adjusted function to change setting entity fields re modification
